### PR TITLE
✨(blog) add Blog post

### PR DESCRIPTION
**Pull request by sveetch**
## Purpose

Richie needs a Blog to add some news without depending on an external application.

## Proposal

This commit adds a new page extension 'BlogPost' with a factory, templates,
basic integration, tests and some page creation from 'create_demo_site'.

## Todo

- [x] Model
- [x] Template
- [x] Placeholders config
- [x] Plugin
- [x] Factory
- [x] Tests
- [x] create_demo_site
- [x] Detail integration
- [x] Getters for listing blogpages from a Category
- [x] Plugin integration
- [x] List integration
- [x] Remember to squash course migration to include blogpost model once every thing is done

### CODE REVIEW

File: tests/apps/courses/test_cms_plugins_blogpost.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        """
+
+        class BlogPostPluginModelForm(forms.ModelForm):
+            """A form for testing the choices in the select box"""
+
+            class Meta:
+                model = BlogPostPluginModel
+                fields = ["page"]
+
+        blogpost = BlogPostFactory()
```
Comment (sampaccoud) [inline tests/apps/courses/test_cms_plugins_blogpost.py]: Add this and the test will fail unless you add the clause I pointed above.
```suggestion
        blog_page = create_i18n_page(published=True)
        blogpost = BlogPostFactory(page_parent=blog_page)
```

File: tests/apps/courses/test_cms_plugins_blogpost.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        other_page_title = "other page"
+        create_page(other_page_title, "richie/fullwidth.html", settings.LANGUAGE_CODE)
+        plugin_form = BlogPostPluginModelForm()
+        self.assertIn(blogpost.extended_object.get_title(), plugin_form.as_table())
```
Comment (sampaccoud) [inline tests/apps/courses/test_cms_plugins_blogpost.py]: As explained in this PR https://github.com/openfun/richie/pull/509/files#diff-67a30e02803ca05e2185d5da408b5847R89 you should count to make sure there are no duplicates:
```suggestion
        rendered_form = plugin_form.as_table()
        self.assertEqual(rendered_form.count(blogpost.extended_object.get_title()), 1)
```

File: tests/apps/courses/test_cms_wizards_blogpost.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        The wizard to create a new blogpost page should be present on the wizards list page
+        """
+        user = UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=user.username, password="password")
+
+        # Let the authorized user get the page with all wizards listed
+        url = reverse("cms_wizard_create")
+        response = self.client.get(url)
+
+        # Check that our wizard to create courses is on this page
```
[the remark this answers is not in the document]
Comment (sampaccoud) [reply tests/apps/courses/test_cms_wizards_blogpost.py]: not resolved yet apprently

File: tests/apps/courses/test_templates_blogpost_detail.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        self.client.login(username=user.username, password="password")
+
+        blogpost = BlogPostFactory(page_title="Preums", fill_cover=True, fill_body=True)
+        page = blogpost.extended_object
+
+        # The page should be visible as draft to the staff user
+        url = page.get_absolute_url()
+        response = self.client.get(url)
+        self.assertContains(
+            response, "<title>Preums</title>", html=True, status_code=200
```
Comment (sampaccoud) [inline tests/apps/courses/test_templates_blogpost_detail.py]: you're not checking that the draft markup is there as announced in the comment?

File: src/richie/apps/courses/models/blog.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    """
+    BlogPost plugin model handles the relation from BlogPostPluginModel
+    to their blogpost instance
+    """
+
+    page = models.ForeignKey(
+        Page,
+        on_delete=models.CASCADE,
+        related_name="blogpost_plugins",
+        limit_choices_to={"publisher_is_draft": True, "blogpost__isnull": False},
```
[the remark this answers is not in the document]
Comment (sampaccoud) [reply src/richie/apps/courses/models/blog.py]: Ok forget these 2 comments, you are not subject to this bug...

**Comment by sveetch** [comment]
### Detail list

blogpost_list

### Detail page

blogpost_detail

**Pull request merged**